diff --git a/docroot/sites/all/modules/contrib/og/og.module b/docroot/sites/all/modules/contrib/og/og.module
index dac0ab7..707278c 100755
--- a/docroot/sites/all/modules/contrib/og/og.module
+++ b/docroot/sites/all/modules/contrib/og/og.module
@@ -3677,42 +3677,42 @@ function og_features_pipe_alter(&$pipe, $data, $export) {
   }
 }
 
-/**
- * Implements hook_migrate_api().
- */
-function og_migrate_api() {
-  $migrations = array();
-  if (db_table_exists('d6_og')) {
-    $migrations['OgMigrateAddFields'] = array('class_name' => 'OgMigrateAddFields');
-    $migrations['OgMigrateContent'] = array('class_name' => 'OgMigrateContent');
-    $migrations['OgMigrateUser'] = array('class_name' => 'OgMigrateUser');
-
-    foreach (node_type_get_names() as $bundle => $value) {
-      $machine_name = 'OgMigrateGroup' . ucfirst($bundle);
-      $migrations[$machine_name] = array(
-        'class_name' => 'OgMigrateGroup',
-        'bundle' => $bundle,
-      );
-    }
-
-    if (db_table_exists('d6_og_users_roles')) {
-      // OG user roles (OGUR) related migrations.
-      $migrations['OgMigrateOgurRoles'] = array('class_name' => 'OgMigrateOgurRoles');
-      $migrations['OgMigrateOgur'] = array('class_name' => 'OgMigrateOgur');
-    }
-  }
-  elseif (db_field_exists('og_membership', 'group_type') && db_table_exists('og') && !db_table_exists('d6_og')) {
-    $migrations['OgMigrateMembership'] = array('class_name' => 'OgMigrateMembership');
-    $migrations['OgMigrateRoles'] = array('class_name' => 'OgMigrateRoles');
-    $migrations['OgMigrateUserRoles'] = array('class_name' => 'OgMigrateUserRoles');
-  }
-
-  $api = array(
-    'api' => 2,
-    'migrations' => $migrations,
-  );
-  return $api;
-}
+///**
+// * Implements hook_migrate_api().
+// */
+//function og_migrate_api() {
+//  $migrations = array();
+//  if (db_table_exists('d6_og')) {
+//    $migrations['OgMigrateAddFields'] = array('class_name' => 'OgMigrateAddFields');
+//    $migrations['OgMigrateContent'] = array('class_name' => 'OgMigrateContent');
+//    $migrations['OgMigrateUser'] = array('class_name' => 'OgMigrateUser');
+//
+//    foreach (node_type_get_names() as $bundle => $value) {
+//      $machine_name = 'OgMigrateGroup' . ucfirst($bundle);
+//      $migrations[$machine_name] = array(
+//        'class_name' => 'OgMigrateGroup',
+//        'bundle' => $bundle,
+//      );
+//    }
+//
+//    if (db_table_exists('d6_og_users_roles')) {
+//      // OG user roles (OGUR) related migrations.
+//      $migrations['OgMigrateOgurRoles'] = array('class_name' => 'OgMigrateOgurRoles');
+//      $migrations['OgMigrateOgur'] = array('class_name' => 'OgMigrateOgur');
+//    }
+//  }
+//  elseif (db_field_exists('og_membership', 'group_type') && db_table_exists('og') && !db_table_exists('d6_og')) {
+//    $migrations['OgMigrateMembership'] = array('class_name' => 'OgMigrateMembership');
+//    $migrations['OgMigrateRoles'] = array('class_name' => 'OgMigrateRoles');
+//    $migrations['OgMigrateUserRoles'] = array('class_name' => 'OgMigrateUserRoles');
+//  }
+//
+//  $api = array(
+//    'api' => 2,
+//    'migrations' => $migrations,
+//  );
+//  return $api;
+//}
 
 /**
  * Implements hook_flush_caches().
diff --git a/docroot/sites/all/modules/contrib/og/og_ui/og_ui.module b/docroot/sites/all/modules/contrib/og/og_ui/og_ui.module
index b265137..edbbb9e 100644
--- a/docroot/sites/all/modules/contrib/og/og_ui/og_ui.module
+++ b/docroot/sites/all/modules/contrib/og/og_ui/og_ui.module
@@ -1171,26 +1171,26 @@ function og_ui_og_ui_get_group_admin($group_type, $gid) {
   return $items;
 }
 
-/**
- * Implements hook_migrate_api().
- */
-function og_ui_migrate_api() {
-  $migrations = array();
-  if (db_table_exists('d6_og')) {
-    $migrations['OgUiMigrateAddField'] = array('class_name' => 'OgUiMigrateAddField');
-    $migrations['OgUiSetRoles'] = array('class_name' => 'OgUiSetRoles');
-
-    foreach (node_type_get_names() as $bundle => $value) {
-      $machine_name = 'OgUiPopulateField' . ucfirst($bundle);
-      $migrations[$machine_name] = array(
-        'class_name' => 'OgUiPopulateField',
-        'bundle' => $bundle,
-      );
-    }
-  }
-  $api = array(
-    'api' => 2,
-    'migrations' => $migrations,
-  );
-  return $api;
-}
+///**
+// * Implements hook_migrate_api().
+// */
+//function og_ui_migrate_api() {
+//  $migrations = array();
+//  if (db_table_exists('d6_og')) {
+//    $migrations['OgUiMigrateAddField'] = array('class_name' => 'OgUiMigrateAddField');
+//    $migrations['OgUiSetRoles'] = array('class_name' => 'OgUiSetRoles');
+//
+//    foreach (node_type_get_names() as $bundle => $value) {
+//      $machine_name = 'OgUiPopulateField' . ucfirst($bundle);
+//      $migrations[$machine_name] = array(
+//        'class_name' => 'OgUiPopulateField',
+//        'bundle' => $bundle,
+//      );
+//    }
+//  }
+//  $api = array(
+//    'api' => 2,
+//    'migrations' => $migrations,
+//  );
+//  return $api;
+//}
