Index: docroot/sites/all/modules/custom/apachesolr_custom/apachesolr_custom.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- docroot/sites/all/modules/custom/apachesolr_custom/apachesolr_custom.module	(revision 1c902e3910720933e87e3fff9a6a9709a4640984)
+++ docroot/sites/all/modules/custom/apachesolr_custom/apachesolr_custom.module	(revision )
@@ -200,7 +200,7 @@
    * any content type that you would like to have promoted in search results. (machine name: field_boost.)
    *
    * At search time, this will multiply the score of the result by 100, so that boosted results always appear
-   * at the beginning of searches. If a view is set to sort by score, then boosted results will appear first.  
+   * at the beginning of searches. If a view is set to sort by score, then boosted results will appear first.
    *
   **/
 
